### DISCUSSION TITLE
Stop labels

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#35] [General]
I am trying to use my own labelled stops. How does one turn off the stop text generated by GOODF?

**Comment by Bumblebee001** [comment]
Also the stops are not activated throughout the image even if mouse settings are set according to size of image.
![image](https://github.com/larspalo/GOODF/assets/6999990/68778d7e-5c56-4618-8f58-88c434de33d0)

**Reply by larspalo** [reply]
To not get any text from GO (or GOODF) set the TextBreakWidth to 0 (zero).

If you want to capture mouse click within the whole image rectangle, you must set the MouseRadius to 0 (zero).

**Reply by Bumblebee001** [reply]
Thanks. I forgot that "trick".... :-/

**Comment by Bumblebee001** [comment]
I also have difficulty in how to represent "preset tabs". Not exactly sure how these tabs on this german organ actually work in reality but there are 2 tabs over each stop....

![image](https://github.com/larspalo/GOODF/assets/6999990/fe343d17-846c-4c75-8745-69df54e00f73)
![image](https://github.com/larspalo/GOODF/assets/6999990/2485f3be-1d66-456f-9e49-667f6c0819ba)

**Reply by Bumblebee001** [reply]
I think the combination pistons (Fr.Comb.I or II) would activate the corresponding row of selected tabs.

**Reply by larspalo** [reply]
Ok, so this is an older type of a free combination system with a mechanical way of setting the combinations. The tabs above the stops are used to set the state of the corresponding stop when the Fr.Comb.I or Fr.Comb.II is activated (that's why you have two rows of extra tabs). The Handreg. would return to reading the state from the larger (normal) tabs. The combinations and Handreg. are mutually exclusive so that only one of them can be active at a time. The Auslöser should disable the current combination without the need to activate another one, if it also at the same time returns the user to the Handreg. or not I cannot say for sure.

Besides the mechanical free combination system, you also have some fixed combinations to take care of.

**Reply by Bumblebee001** [reply]
Can you spare a moment to guide me briefly what I have to do? I'm a bit mentally exhausted working on this sample-set and cannot think..... not to mention that my ears still buzz after listening to all those loops!

**Reply by Bumblebee001** [reply]
So far:
![image](https://github.com/larspalo/GOODF/assets/6999990/381f2c69-d269-4a2d-b711-88658005247f)

**Reply by larspalo** [reply]
The really tricky part is how to solve the mutually exclusive part, where pushing one button should visibly disable the other, not just activate back end stops from a set of selected buttons (tabs). I wonder if that perhaps could be done by using completely transparent on/off images for the activating switches that are overlayed on top of underlying intermediate switches (with visible images) that only show the state that's actually controlled by the transparent switches? Still it's a tricky setup of the switch network... I must think about it some more.

**Reply by Bumblebee001** [reply]
So it's not just me unable to figure it out..... :-D

**Comment by larspalo** [comment]
I've figured out one possible way to do a model of this type of old combination system. There might be other ways and if someone figures out an easier way, please make it public!

My solution is a combination of a switch network and protected generals to work around the switch referencing limitations of GO. The system is somewhat difficult to explain so I attach a schematic image of how the objects interact.

![OldCombinationsDrawing](https://github.com/larspalo/GOODF/assets/6173941/291c69b9-c481-4799-9240-52be1b6b0f13)

The intermediate switches for Handreg, Comb I and Comb II that actually can take Input are optional, but I included them in my example to avoid displaying them as they are clickable. I also add a working sample set below that can be examined and tested.

[OldComb.zip](https://github.com/larspalo/GOODF/files/12138714/OldComb.zip)

**Reply by larspalo** [reply]
By the way, while testing this I found a few other bugs in GOODF that I'll try to fix. One is about the button placement and the other about that (some of) the default values for a button seems to revert unconditionally if a built in bitmap is used which makes adjustments not possible.

**Comment by Bumblebee001** [comment]
Yes.... I think that's how they are supposed to work. Now that has to be applied to all the stops, couplers and Tremolo. Would a cancel button for each row be a useful addition? They do not seem to be present on the real organ but then I don't understand the German labels on this instrument.

**Comment by larspalo** [comment]
Such combination systems are way older than me... I've played on only a few instruments with similar systems so I really cannot say how that instrument is supposed to behave. I'd guess that the "Tutti" would "override" any other combination or hand register - quite likely without visibly changing the states of any of them - you'd only see that the "Tutti" would be pushed! The same would go for the "Zungen ab" (Reeds off) - it would likely disable any active reeds without changing the state of the tabs for Handreg. or Comb I or II. The "Cresc. ab" is likely a button to disable the swell (register crescendo or rollschweller, not to confuse with the enclosure). What the "Ped. Comb." does I cannot say - is there possibly a special combination for the pedal only?

Then the "Auslöser" remains. From what I've read it's a way of cancelling the current combination without activating another one. Some sources seem to indicate that it actually returns to the last used combination, but I cannot say for sure without sitting at the organ and testing it.

If you have the possibility to go to the organ and test how they behave that would be the best - if you can get someone else to do it and explain to you how they work - that's the next best. If neither is possible, you'll just have to make a decision on how you want the virtual organ to be based on what you know (and want).

**Reply by Bumblebee001** [reply]
Luckily I was with an English organist friend when I recorded the organ and I know the Maltese organbuilder who installed it, himself an accomplished concert organist. So despite me not knowing much about this organ there are sources I can get information from. I'll work on that now that I have completed most of the basic work for the set. I do want to get the tabs on next then I'll proceed to the rest.

**Comment by Bumblebee001** [comment]
This is the console of the real organ:
![image](https://github.com/larspalo/GOODF/assets/6999990/90c326dd-a454-47dc-8f53-32b0fcefbb12)

**Reply by Bumblebee001** [reply]
As you see there is a third row of ped combiniations possible which can be selected to over-ride any other selections (hand or I or II).

**Reply by larspalo** [reply]
Ok, right. They seem to be a later add on though, as the style is very different from the others. But it should be possible to just add a third combination affecting/activating only them. There are some extra tabs to the left too that I cannot read what they are supposed to do. It seems to be an pneumatic action (at least stop action) organ. Cool!

**Comment by Bumblebee001** [comment]
I am in the process of copying what you did with 3 stops and applying to 35 stops! It is a big task and I have had to use Notepad++ to facilitate part of the task and edit (/replace) text and digits accordingly. Is there a way this bulk processing can be achieved with GOODF?  Currently only one item is selectable. It would help if several items can be selected and the changes or additions would be applied to all the selected items leaving one to just add additional things that would be stop specific.

**Reply by larspalo** [reply]
It's possible to copy properties of GUI Elements to others either on the same panel or on another (done by right clicking the "source" element). In the "structural" part it's not (yet) possible except for a few features (copying pipes and such). What exactly would you like to be able to do?

**Comment by Bumblebee001** [comment]
Closing the file (even though no changes were made) gives this:
![image](https://github.com/larspalo/GOODF/assets/6999990/19ced6a8-7dd8-4408-b8c5-d11ada46cee9)

I had to force close the program from Task Manager

**Reply by larspalo** [reply]
It's hard to guess what went wrong. If you manage to repeat the same crash, please open an issue with exact steps to re-produce it.

**Comment by Bumblebee001** [comment]
It would help to add "Remove All" ....

**Reply by larspalo** [reply]
Ctrl+N is a new organ. More remove all than that is hard to find... :-)

Anyway, only the first object of a type need confirmation. So to remove for instance all switches you can select the last and remove that and then keep clicking the remove button until all objects of that type is removed.

**Comment by Bumblebee001** [comment]
Now: the next headache begins....... you created a "Behind the screne" panel with all those stops for just 3 sample stops...... I have 35!!!! :'-(

PS: the text on those stops are far too small to read. :-(

**Reply by larspalo** [reply]
The "Behind the scene" was just for you to see how the objects interact. You don't need to include that for your sample set.

**Comment by Bumblebee001** [comment]
Applying the options for one stop to all the rest as you have done with
just 3 stops.

**Reply by larspalo** [reply]
What options?

**Reply by Bumblebee001** [reply]
![image](https://github.com/larspalo/GOODF/assets/6999990/f8b3298f-9f78-4156-826e-dc69044124dc)

I managed using Notepad++.... but that's not what is supposed to be done..... one should not rely on Notepad++ to get some things done...

**Reply by larspalo** [reply]
Ok, so basically just another way of distributing the switches to many elements at once instead of doing it one at a time. That's a possible future improvement.

**Comment by Bumblebee001** [comment]
Thank God for that!!!!

**Comment by Bumblebee001** [comment]
So how do I add the graphics for the tabs?

**Reply by larspalo** [reply]
Just make one GUI element tab like you want them all to be - then right click on it and select all other similar elements you want to be the same and confirm the copy operation.

**Reply by larspalo** [reply]
Have you watched any of the screencasts I've made as help on how to use GOODF? https://www.youtube.com/playlist?list=PLNz9AJhjubTdw0pV2nZC_XfphJhJV5ly3

Many things are very basic but in the third with Builtin options at around 2:20 I show how to copy things for GUI elements.

**Comment by Bumblebee001** [comment]
I think I understand. Will revert back to you if I don’t succeed.

**Comment by Bumblebee001** [comment]
I think I would need to replace the current stops with the respective hand
switches I’m guessing?

**Reply by larspalo** [reply]
Yes, that's likely. But it's possible to leave them for now, create the switch that will be displayed instead first, add it to the panel, and copy the attributes of each stop to corresponding switch before removing the stop.

**Reply by Bumblebee001** [reply]
I added the new elements yet when I go to the old stops and right click to copy attrubutes, the new elements do not appear in the options!!!

**Reply by Bumblebee001** [reply]
I saved. Closed. Reloaded. No change

**Reply by larspalo** [reply]
Right, that was implemented with bad7bcdff69830e5e3d9c9594aa950efe5f4dd4d after 0.6.2 and isn't available yet if one doesn't compile.

**Comment by Bumblebee001** [comment]
I just tried closing the file I was working with. That’s all. Like I do
with other software.

**Reply by larspalo** [reply]
Well, I've found one crash causing issue with the wxTreeCtrl but I don't know if that was the cause in your case.

**Comment by Bumblebee001** [comment]
Ok, thanks for the update.

**Comment by larspalo** [comment]
@Bumblebee001 Another way of implementing the Tutti that will not visibly change the states of the Handreg., Comb I or Comb II tabs (but cancel the H, I and II of course): [OldStyleCombinationsTutti.zip](https://github.com/larspalo/GOODF/files/12164214/OldStyleCombinationsTutti.zip)

Drop it in the same folder as the example above to test it.

**Reply by Bumblebee001** [reply]
I got it to work such that all the desired tabs become visibly on too. 

In the organ I am working with, the organbuilder/installer mentioned that two stops are *not* included in the tutti, This was easy to achieve by removing the two stops from the list of stops to be activated by the T piston.

**Reply by larspalo** [reply]
> I got it to work such that all the desired tabs become visibly on too.

If you're happy with that - it's the easiest way. But I believe that's not how the original organ behaves.

> that two stops are not included in the tutti,

Sure, that's easy in my version above too. Just remove the tutti reference in the stops that should not be activated.

**Comment by larspalo** [comment]
Here's a version how I believe the Reeds off should work for this type of organ/combination system, also updated the Tutti in this version and showed how to exclude one stop from it.

[OldStyleCombinationsReedsOff.zip](https://github.com/larspalo/GOODF/files/12175697/OldStyleCombinationsReedsOff.zip)

Drop it in the same folder as above to test it.

**Comment by larspalo** [comment]
For your reference, here's a version that has a special Ped. Comb. for stop 3 that will, when activated, override any other combination (for that stop only in the example, but in your case it will of course be necessary for all the pedal stops). 

[OldStyleCombinationsPedComb.zip](https://github.com/larspalo/GOODF/files/12237796/OldStyleCombinationsPedComb.zip)

Drop it in the same folder as above to test it.

**Comment by larspalo** [comment]
An almost working attempt at implementing crescendo. For one reason or another, sometimes the Cresc. cancels the state it should be falling back to, so it's not 100% foolproof right now. But it should get you started.

[OldStyleCombinationsCresc.zip](https://github.com/larspalo/GOODF/files/12261157/OldStyleCombinationsCresc.zip)

Drop it in same folder as above and extract it to test it. The programming procedure is somewhat complicated but I've included a pre-programmed .cmb that will auto load with the .organ file like you should do.

For the programming step I had to make quite a few changes manually like setting the GO crescendo override mode so it wouldn't change any already existing elements, then manually de-activate all other inputs (which you could actually have shown on the programming panel but I had them already on the behind the scene so I just used that). First after that it's possible to start the programming procedure on the special panel for that... I don't have time to work on this so much more at the moment so you're on your own to figure out possible improvements.

**Reply by Bumblebee001** [reply]
The problem with this is that as soon as you hit the cresc pedal, the Cresc Ab piston becomes activated effectively resulting in the cresc being active all the time and is never or can never be turned off.

**Reply by larspalo** [reply]
Yes, that's kind of how I thought it should work. As soon as you touch the crescendo pedal it gets active and to return from the crescendo to H, I, II or whatever one press the Cresc. Ab button. At least in theory... As sometimes it seems to require actually pressing H, I or II, but it's not consistent.

It shouldn't be impossible to change that behavior though if that's desired. But I leave that decision up to you.

**Comment by Bumblebee001** [comment]
What you say makes sense in a way but what the organ actually does in
reality is something I’ll need to verify. Meanwhile I’m creating the
switches….. sigh!!!

**Reply by larspalo** [reply]
If it's of any comfort, I feel with you... But on the bright side: your sample set is the first I know of that has such a sophisticated old style combination setup! I'm mainly working on older instruments without any combinations, or on newer instruments that are much more modern and behaves more like GO does out of the box so to speak.

**Reply by Bumblebee001** [reply]
Lars.... I followed your example and applied as always to the rest of the stops. I created the cresc panel and set of switches etc etc. The crescendo does not produce any sound even though the switches become engaged as programmed on this screen (not the main panel). Also for H, I, II to work I have to press the cresc ab otherwise those won't play either. I'm confused. I tried applying default emgage to Cresc ab then to Cresc (NOT) separately..... problem remained. Where could I be going wrong? The cmb generated is in the sample-set folder and I deleted all cache entries and have no other cmb that could be loaded other than the latest one I created.

**Reply by Bumblebee001** [reply]
The only thing I left out is the Cresc Ab on the programming screen. Is it important? 
How would it effect the performance of the sample-set?
I have it on the main panel though...

Also I have not created "Behind the Scene" panel as you did.

**Reply by larspalo** [reply]
I don't even remember the exact programming details since yesterday (too many other things going on at the same time right now), but I think that I had to manually disable all the inputs of H, I, II etc, while programming the Cresc (and change the state of Cresc. Ab) so it could be a good idea to have them on that panel (since you don't have the purely academic "Behind the Scene" panel).

**Reply by Bumblebee001** [reply]
I forgot to mention that trying to use the cresc pedal cancels all registration (turns off all stops).

I added the Cresc Ab to the programming screen - no change

**Reply by Bumblebee001** [reply]
I just noticed that I forgot to add the Cr selector to the stops in the manuals...... silly me!!!! I was too much focused on the switches and forgot all about the actual stops!

**Reply by larspalo** [reply]
Actually, thinking about this, maybe the crescendo itself would benefit from being setup in a scope/scoped mode instead of regular... Not that I've tested it. But it make sense to only include the programmed stop switches for Cr in the crescendo and not any other elements. I think I kind of solved this in the example by changing the crescendo override mode from override to add before starting to program. I've not implemented that option yet in GOODF as it's a fairly new change in GrandOrgue itself.

Drawback of scope/scoped could be that one perhaps loose the automatic on feature when using the crescendo swell. Unfortunately, this would need much more testing than I have time for.

**Reply by Bumblebee001** [reply]
Ok..... sound Ok - Cresc now works BUT the selected registration on the H, I and II cancel/ turn off as soon as the cresc pedal is activated. The should remain whilst the cresc over-rides and ocne the cresc ab is pressed, the last registration should resume where left off.

**Reply by Bumblebee001** [reply]
You said: 
I think I kind of solved this in the example by changing the crescendo override mode from override to add before starting to program. 

Can you explain that? What should I do?

**Reply by larspalo** [reply]
Click the rightmost button on the builtin crescendo panel of GO before you start programming.

**Reply by Bumblebee001** [reply]
I tried removing an image that I created. Whilst a window appeared that allowed me to select a blank entry, the off switch side remains there and cannot be removed!

![image](https://github.com/larspalo/GOODF/assets/6999990/c2d14272-0002-4379-8c3e-eed3e9035ce2)


![image](https://github.com/larspalo/GOODF/assets/6999990/fea6b21c-9bd1-41f7-9a0d-c1c480892961)

**Reply by Bumblebee001** [reply]
I think a good option is to allow direct text entry and deletion.

**Reply by Bumblebee001** [reply]
If done in the reverse order than it is possible to remove both..... starting from the off switch image - then the on switch image.

**Reply by larspalo** [reply]
> the off switch side remains there and cannot be removed

Yes, that's indeed a bug that needs fixing. The question is just how to best do that? One possibility, _which I'm leaning towards_, is to detach the image/mask on and image/mask off operations completely so that they are totally independent from each other. (Even in GO I don't think that there is any enforcing of having them both customized, it's perfectly possible to use the builtin bitmaps for just one of the states - but the bitmap size of off must match what is used for on). Another possibility is to empty both fields if the image on is emptied, either directly without asking or after another question dialog asking if that is what the user desires and if so check that the off image will match the default on image size (and if it doesn't just remove it too). In any way, the handling of on and off states need to be separated.

> I think a good option is to allow direct text entry and deletion.

I don't. It's opening a can of worms! User input can frequently be wrong, not to mention that it can easily lead to portability problems as the windows file system is case-insensitive so a faulty entry could work on windows while it will not on any case sensitive system. Therefore it's best to only allow the paths to be filled in by selection from the actual file system and never by hand. The files _need_ to exist and have a full path exactly as they are named, not only for GOODF but more importantly for GO, which means that no faulty user input is acceptable at all.

**Comment by Bumblebee001** [comment]
Using the override option is producing unexpected behaviour. 

When Overridden is ON and Ab ON: H registration works. I, II and T are dead. Use of cresc good but registration goes off and applying new reg H, I, II don't play at all irrespective of Ab on/off.

When Overridden is OFF and Ab ON only the H works (provided the cresc is not at 1). Piston dead completely. Cresc works.

It is to be noted that I programmed the cresc as follows:
1 - no active stops ............................. in this state the H does not work at all
2-4 - each set the same (step 1 of registration chart)
5-7 - each set the same (step 2)
etc
until 30-32 - Tutti incl the extra stops I created (II-II 16, II-II 4, I-I 16 and I-I 4) 

so apart from stage 1, changes in cresc occur every 3rd stage (10 steps of registration spread over 32 possible cresc stages)

I hope I got all that right!

**Reply by larspalo** [reply]
I initially had a little issue with that too, but when I programmed the crescendo correctly it sorted itself out.

> (10 steps of registration spread over 32 possible cresc stages)

Yes, I think that's a good idea to have the full range of operation of the pedal.

**Comment by Bumblebee001** [comment]
Well... I recreated the cresc with the override on and I notice it's the pedal piston that is deactivating the manuals (H, I, II)....

**Reply by Bumblebee001** [reply]
What if I set the steps of the crescendo as protected generals and somehow make the pedal simply go from one general piston to the next? I have already created 5 generals that you advised me to delete..... , and still have them, and I can create another 5 for this purpose? Maybe it was a good idea to have them....... though they would now serve a different but specific purpose that's giving us so much trouble....

**Reply by Bumblebee001** [reply]
The pedal will activate a "sequencer" going from general 1 to general 10 (hidden) and back accordingly. The Cresc Ab piston would have to be able to activate and deactivate this sequencer.

**Reply by larspalo** [reply]
Honestly, I didn't even know that the crescendo could activate generals at all (which I kind of doubt since the crescendo steps actually are generals in themselves)... I don't even know exactly how I managed to get the example working as I'm doing much too many things right now. You could try setting the global CombinationsStoreNonDisplayedDrawstops=N and see if that makes any difference in your case.

**Reply by Bumblebee001** [reply]
![image](https://github.com/larspalo/GOODF/assets/6999990/cf4c6b3e-9dd8-42e4-9dc9-e598689b987c)

**Reply by Bumblebee001** [reply]
Can you at least tell me how to **prevent** the Pedal Piston from getting activated when the crescendo is used? 

At the same time whichever of these are on (H, I or II) must not go off when the crescendo is used.

**Reply by larspalo** [reply]
![CombStoreNonDisplayed](https://github.com/larspalo/GOODF/assets/6173941/c2ed3a33-3c43-4dc5-98a4-c9282f9673a3)

> prevent the Pedal Piston from getting activated when the crescendo is used

Most likely by making sure no one of them are active while you're programming the crescendo steps. That's one of the reasons all kinds of inputs might be good to include on the crescendo programming panel if you don't have a Behind the scene where they can be accessed and checked for state.

Anyway, as I said before this sample set of yours might be a case where setting the scope (limiting to only relevant switches) for the crescendo steps and then storing the combinations for it as scoped might be a good alternative solution.

**Reply by Bumblebee001** [reply]
I am not succeeding.
I set Scope, then Set, then next (2), then select the switches (stops), click current, next (3), current, next (4), current, next (5), add other stops, current, next (6), etc..... at the end scoped and unclick save. Nothing is saved!

PS Yes.... I watched your video but the demo was for setting pistons and then supposedly setting to hardware (listen to event) which the viewer does not get to see and therefore unsure what you are doing......

**Reply by Bumblebee001** [reply]
At best the stops come on on pressing the pedal but do not regress when crescendo pedal is raised.

**Reply by Bumblebee001** [reply]
I forgot to change that nondisplayed drawstop setting. I'll try again

**Reply by Bumblebee001** [reply]
Nope...... crescendo works up but not down! I give up!

**Reply by larspalo** [reply]
> I am not succeeding.
I set Scope, then Set, then next (2), then select the switches (stops), click current, next (3), current, next (4), current, next (5), add other stops, current, next (6), etc..... at the end scoped and unclick save. Nothing is saved!

If you had watched carefully and drawn a few conclusions, you'd see that it's a two step process. You're programming the crescendo swell so you cannot do exactly like in the screencast since you don't have access to the buttons for each combination directly.

First you set the scope for all the steps:

1. Make sure only **all** the _switches (cr)_ and possibly the cresc ab is active. (This is important, all elements possible/needed to affect in the next programming stage should be active at this point!)
2. Select "Scope", then "Set"
3. From step 1 hit "Current"
4. Then "Next" until you reach step 32.
5. Deactivate "Set"

In the second step you do the actual programming of the steps:

1. Back step down to 1
2. Select "Scoped"
3. Activate "Set"
4. Activate the switches you want on for step 1
5. Hit "Current"
6. If you want the same combinations for next few step(s) too, hit "Next"
7. Activate switches for next step to store
8. Hit "Next" until you're ready to program next combination change
9. Repeat (7-8) until all combinations are set for all the steps
10. Deactivate "Set"

Test the functionality. Depending on exactly how you want the setup to work, and depending on how you've implemented the Cresc ab, purely as switch or as a general, you might want to include additional elements into the scope above and then in the programming phase you make sure they are de-activated for each programmed step. Experiment, and think about exactly what you need to happen in order to accomplish the task at hand.

**Comment by Bumblebee001** [comment]
I am lost because I don’t know what the scope, scoped, current and next/
previous buttons do…. what their properties are. I understand what set does
but I thought I knew what the others did based on the name but it seems
there is more to them than what looks obvious.

Is the purpose of the first stage of the process to prepare the “numbers”
as to which elements each is to expect or memorize the selections in the
second stage?

Which one saves or assigns the selected stops to a particular step (number)?

I will attempt again trying to follow the instructions but it’s like
telling a blind person turn left now walk two paces and turn right without
that person being aware of the layout of the space around him and his
orientation within that space.

**Reply by larspalo** [reply]
The whole purpose of the first step is to tell the crescendo that only the selected elements (switches in your case) can be affected by the crescendo, ie. turned on or off. This is done to avoid having other switches cancelled by the crescendo. In principle - this is what setting the override mode to add should have done too.

"Current" stores the currently active elements into the current combination (crescendo) step. "Next" advances the step to the next number, and if "Set" is active it will also store the currently active elements into the combination (crescendo) step you arrive to.

The purpose of the second step is to actually program into each combination (crescendo) step the desired elements (switches in your case).

**Comment by Bumblebee001** [comment]
Also the cresc ab piston has to be off during the entire use of the cresc
pedal. It is turned on only to turn the entire cresc off whatever stage it
is in, whether fully down or partially so. Upon turning off the previous
registration should kick in /become active automatically resuming where it
was before the cresc was used.

**Reply by larspalo** [reply]
I'm not completely sure how to do it best, but you can try to display the Cresc ab in inverted state (as I did if I remember correctly) and use its not switch to control other elements. If the Cresc ab then is programmed active into the crescendo steps it will display as if it was not active and possible to press. When the user then presses the Cresc ab it will actually de-activate (but be displayed as active == pushed) and no longer activate the crescendo selectors for the stop(s) and allow any other combination to pass through instead.

**Reply by Bumblebee001** [reply]
I must have programmed the crescendo pedal correctly as the stops become activated as expected (except that the first step, No.1, which I supposed to have set as all off, did not save). The Cresc Ab is off/on in reverse and the H, I, II do not function at all neither do they show that they have changed state whether Cresc Ab in on or off.

So still not working correctly!

**Reply by Bumblebee001** [reply]
No cmb file generated; only cache file created.

**Reply by Bumblebee001** [reply]
Tutti also not functional; and the Ped Comb. allows cresc to work only when off.

**Comment by larspalo** [comment]
> No cmb file generated; only cache file created.

Are you saying that when you in GrandOrgue go to File->Export Settings choose the correct directory and write a file name that match the .organ file without the organ extension and click on the file dialog ok/save button that no .cmb file is created? Or are you just doing it wrong?

As before, I have no idea why it's not working for you. I attach a scoped version of the crescendo sample that seem to be working exactly as it should and perhaps even a bit more robust than the previous version. Drop it in the same directory as above and extract it to test.

[OldStyleCombinationScopedCresc.zip](https://github.com/larspalo/GOODF/files/12299508/OldStyleCombinationScopedCresc.zip)

**Reply by Bumblebee001** [reply]
Exporting does generate a cmb..... Bit before I don't remember me having to do this to get a cmb file created. Well.... exporting ok.

**Reply by Bumblebee001** [reply]
With the two files I generated by export combinations and export settings (cmb and another yaml), GO crashed and will not load. I had to delete both files then GO opened and I was able to load the organ back!

Nothing coming easy! 

I'll look at your example.
